src/main/java/com/treasure/hunt/view/CanvasController.java

```diff
@@ -50,29 +82,127 @@ void drawShapes() {
                 gameManager.get().getGeometryItems(true).forEach(geometryItem ->
                         geometryItem.draw(graphics2D, shapeWriter)
                 );
+                if (this.highlighter != null) {
+                    this.highlighter.draw(graphics2D, shapeWriter);
+                }
             }
         });
     }
 
+    /**
+     * This clears the {@link GeometryItem}'s from the canvas.
+     */
     private void deleteShapes() {
         if (gameManager == null) {
             return;
         }
         canvas.getGraphicsContext2D().clearRect(0, 0, canvas.getWidth(), canvas.getHeight());
     }
 
+    /**
+     * Executed when the mouse was pressed and released.
+     * TODO To much game logic in here!!11 maybe wrap into GameManager
+     * Executing this will select a {@link GeometryItem} nearest to the mouse position,
+     * with a maximum distance of {@link CanvasController#MOUSE_RECOGNIZE_DISTANCE}.
+     *
+     * @param mouseEvent corresponding {@link MouseEvent}
+     */
     public void onCanvasClicked(MouseEvent mouseEvent) {
         if (gameManager == null) {
             return;
         }
+
+        /*
+         * Only execute this (selecting a GeometryItem),
+         * when the canvas will not be moved.
+         */
+        if (dragged) {
+            dragged = false;
+            return;
+        }
+        dragged = false;
+
+        Vector2D mousePositionInGameContext = dragStart.subtract(offsetBackup);
+        mousePositionInGameContext = mousePositionInGameContext.divide(transformation.getScale());
+
+        GeometryItem geometryItem = gameManager.get().pickGeometryItem(
+                new Coordinate(mousePositionInGameContext.getX(), -mousePositionInGameContext.getY()),
+                MOUSE_RECOGNIZE_DISTANCE / transformation.getScale());
+        if (geometryItem != null) {
+            Geometry geometry = geometryItem.getGeometry();
+            log.info("recognized: " + geometry); // TODO delete
+            this.selected = geometryItem;
+            if (geometryItem.getGeometry() instanceof Point) {
```

Comment:
I don't like what we do there in the end we need a highlighter for every Geometry???

Reply:
This is only for demonstration, that geometryItem-selection works.
Of course, we could do other things